DemoApp/src/Console.cc

```diff
@@ -25,13 +28,56 @@ size_t getConsoleHeight() {
     CONSOLE_SCREEN_BUFFER_INFO sbInfo;
     GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), &sbInfo);
     return sbInfo.dwSize.Y;
-#elif defined __unix__
+#elif defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
     struct winsize size {};
     ioctl(STDOUT_FILENO, TIOCGWINSZ, &size);
     return size.ws_row;
 #endif
 }
 
+#if defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
+void setFlag(tcflag_t flag, bool state) {
+    struct termios tios {};
+    tcgetattr(0, &tios);
+    if (state) {
+        tios.c_lflag |= flag;
+    } else {
+        tios.c_lflag &= ~flag;
+    }
+
+    tcsetattr(0, TCSAFLUSH, &tios);
+}
+#endif
+
+void enableCanon() {
+#if defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
+    setFlag(ICANON, true);
+#endif
+}
+
+void disableCanon() {
+#if defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
+    setFlag(ICANON, false);
+#endif
+}
+void enableEcho() {
+#if defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
+    setFlag(ECHO, true);
+#endif
+}
+
+void disableEcho() {
+#if defined(__unix__) || defined(unix) || defined(__unix) || defined(__linux__) || \
+    (defined(__APPLE__) && defined(__MACH__))
+    setFlag(ECHO, false);
+#endif
+}
+
```

Comment:
Czy te funkcje mają nic nie robić na windowsie? Czy chat wtedy zadziała?

Reply:
Nie mam pojęcia na ten moment jak to zrobić na windowsie, nie wiem o metodach na uzyskanie takiego efektu, planuję szukać dalej, ale na ten moment staram się, żeby się chociaż kompilowało na Widowsie. Po prostu na Windowsie wiadomość pisana, ale jeszcze nie skończona będzie mogła zniknąć z ekranu (gdy przyjdzie nowa wiadomość) i użytkownik będzie musiał ją dokończyć nie widząc tego co napisał wcześniej.